### PR TITLE
Add line spacing support to MSW wxTextCtrl, expand wxTextAttr docs

### LINKED ISSUE #24045: Line spacing for wxTextAttr for a wxTextCtlr under MSW has no effect. Is this expected?
Line spacing for `wxTextAttr` for a `wxTextCtlr` under MSW has no effect.

### Description

I create a `wxTextCtrl` with RICH2 support, but when I apply a `wxTextAttr` with line spacing, the line spacing has no effect:

```
m_style.SetLineSpacing(wxTEXT_ATTR_LINE_SPACING_TWICE);
m_textEntry->SetDefaultStyle(m_style);
m_textEntry->SetStyle(0, m_textEntry->GetLastPosition(), m_style);
```

The docs don't mention this not being supported for `wxTextCtrl`, but I'm wondering if this is by design. Looking at "msw/textctrl.cpp," there is no mention of line spacing at all (just spacing before/after paragraphs, which is a different flag).

I believe `PARAFORMAT2::bLineSpacingRule` is the field to control this (https://learn.microsoft.com/en-us/windows/win32/api/richedit/ns-richedit-paraformat2). I would be happy to make a patch for this, but wanted to confirm that this isn't by design first.

### Platform and version information

- wxWidgets version 3.3
- wxMSW

### PR DESCRIPTION
**Pull request by Blake-Madden**
This should close #24045.

I expanded the `wxTextAttr` docs, mentioning which platforms support which features for `wxTextCtrl`.

Also, `wxTEXT_ALIGNMENT_JUSTIFIED` actually is supported for MSW for `wxTextCtrl`, so I explained that.

I omitted discussing macOS because from what I can tell, `wxTextAttr` is not used at all for `wxTextCtrl`. Looking at the code, that seems to be the case. However, that seems like a bold statement to make here without others confirming that. If that can be confirmed, I'll explain that too.

For line spacing under GTK+, you have to use `pango_layout_set_line_spacing()` (I see no analog via GtKTextView other than line-space functions that only deal with pixels). That proved too complicated for me. Sorry, I was hoping to add line spacing support for wxGTK here also, but that will have to be another day.

### CODE REVIEW

**Comment by vadz** [comment]
Mac does have support for at least some attributes, see `wxNSTextViewControl::SetStyle()` in `src/osx/cocoa/textctrl.mm`.

**Review by vadz — COMMENTED** [review]
Thanks for the change and the docs updates but it would be great if this could be improved a bit further. TIA!

**Comment by Blake-Madden** [comment]
> Mac does have support for at least some attributes, see `wxNSTextViewControl::SetStyle()` in `src/osx/cocoa/textctrl.mm`.

Of course, I totally forgot this was Objective-C.  I've made the other changes, but will go through this file and document what macOS supports...

File: src/msw/textctrl.cpp [hunk 1]
```diff
@@ -3179,6 +3179,13 @@ bool wxTextCtrl::MSWSetParaFormat(const wxTextAttr& style, long start, long end)
     }
 
 #if wxUSE_RICHEDIT2
+    if ( style.HasLineSpacing() )
+    {
+        pf.dwMask |= PFM_LINESPACING;
+
+        pf.bLineSpacingRule = ((style.GetLineSpacing() == wxTEXT_ATTR_LINE_SPACING_NORMAL) ? 0 :
+                               (style.GetLineSpacing() == wxTEXT_ATTR_LINE_SPACING_HALF) ? 1 : 2);
```
[the remark this answers is not in the document]
Comment (Blake-Madden) [reply src/msw/textctrl.cpp]: Done. I did **NOT** use a `default`, so `-Wswitch` should pick up on undefined branches then.

**Comment by Blake-Madden** [comment]
OK, everything should be documented now (including a couple new features in this PR).

**Comment by vadz** [comment]
Great, thanks! I'll (squash -- unless you want to preserve the commits?) merge this soon and could backport it to 3.2 too if you'd like.

**Comment by Blake-Madden** [comment]
> Great, thanks! I'll (squash -- unless you want to preserve the commits?) merge this soon and could backport it to 3.2 too if you'd like.

Squash away :)  Sorry, I did this more piecemeal than I should have.

3.2 backport sounds good to me, if not too much trouble.

**Pull request closed without merging**